lib/paginator/ecto/query/dynamic_filter_builder.ex

```diff
@@ -0,0 +1,51 @@
+defmodule Paginator.Ecto.Query.DynamicFilterBuilder do
+  @dispatch_table %{
+    desc: Paginator.Ecto.Query.DescNullsFirst,
+    desc_nulls_first: Paginator.Ecto.Query.DescNullsFirst,
+    desc_nulls_last: Paginator.Ecto.Query.DescNullsLast,
+    asc: Paginator.Ecto.Query.AscNullsLast,
+    asc_nulls_last: Paginator.Ecto.Query.AscNullsLast,
+    asc_nulls_first: Paginator.Ecto.Query.AscNullsFirst
+  }
+
+  @callback build_dynamic_filter(%{
+              direction: :after | :before,
+              entity_position: integer(),
+              column: term(),
+              value: term(),
+              next_filters: Ecto.Query.dynamic() | boolean()
+            }) :: term()
```

Comment:
dialyzer is not happy using `Ecto.Query.dynamic()`, because that is exported as `@opaque`. That is why I had to use `term` here. If anyone knows of a better way to describe this function, please let me know 🙇🏽